File: LeetCode Solutions/Problems/2379. Minimum Recolors to Get K Consecutive Black Blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int countW = k;
        int n = blocks.length();
        for (int i=0;i+k-1<n;i++){
            int count = 0;
            for (int j=i;j<i+k;j++){
                char ch1 = blocks.charAt(j);
                if (ch1 == 'W' ){
                    count++;
                }
            }
            countW = Math.min(count, countW);
        }
        return countW;
    }
}

File: LeetCode Solutions/Problems/2379. Minimum Recolors to Get K Consecutive Black Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        min_operation, step = 0, 0
        while (step < len(blocks) - k + 1):
            temp_arr = blocks[step : step + k]
            if step == 0:
                min_operation += temp_arr.count("W")
            else:
                min_operation = min(min_operation, temp_arr.count("W"))
            step += 1
        return min_operation